### ISSUE TITLE
Erroneous indentation warning in template literals contents

### ISSUE THREAD

**Issue by mercmobily** [eslint/eslint#10932]
**Tell us about your environment**
* **ESLint Version:** 5.6.1

* **Node Version:** 8.12.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

    module.exports = {
        "env": {
            "browser": true,
            "es6": true
        },
        "extends": "eslint:recommended",
        "parserOptions": {
            "ecmaVersion": 2018,
            "sourceType": "module"
        },
        "rules": {
            "indent": [
                "error",
                2
            ],
            "linebreak-style": [
                "error",
                "unix"
            ],
            "quotes": [
                "error",
                "single"
            ],
            "semi": [
                "error",
                "never"
            ]
        }
    };

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
class ShopCart {
  render () {
    const { html, map } = this
    return html`
      <style>
        :host { display: block; }
      </style>
      ${map((item) =>
        item++
      )}
    `
  }
}

new ShopCart 

```

```bash
./eslint.js -c config.js test.js
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

I expected it to go through without any warnings. However, I got this output:

```bash
/home/merc/.atom/packages/linter-eslint/node_modules/eslint/bin/test.js
   9:1  error  Expected indentation of 4 spaces but found 8  indent
  10:1  error  Expected indentation of 2 spaces but found 6  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```
Please note that the right indentation for that `item++` is 8, not 4. Same applies to the closing of the map() function.

**Comment by mercmobily** [comment]
Anyone...? I am especially interested if one of the developers could point out a previous version that isn't affected by this problem, so that I could use it...

**Comment by platinumazure** [comment]
Hi @mercmobily, thanks for the issue and sorry for the delayed reply.

I think the (new/post-4.0.0) indent rule has always worked this way. Given that the rule aligns by tokens and we should be able to track the location of the `${` tokens, though, I would be curious to see if we could enhance the rule to tolerate both configurations somehow.

**Comment by not-an-aardvark** [comment]
I think this is a duplicate of https://github.com/eslint/eslint/issues/10814.

**Comment by mercmobily** [comment]
@platinumazure You know much better than me. But... since `${...}` contains code, and the code is meant to be indented by 2 spaces, shouldn't this be marked as a bug?

Right now, well formatted code is screaming out "this is wrong", but the advice given is actually wrong...

**Comment by mercmobily** [comment]
(What makes this problem more of a problem, is that it's not even possible to disable the warning of the nested template... So, right now I am effectively stuck with the problem!

**Comment by platinumazure** [comment]
@mercmobily You can use the [`ignoredNodes` option](https://eslint.org/docs/rules/indent#options) of indent to disable the warning in this case.

I've been trying to create a demo of this option, but the demo is being flaky today and I'll have to investigate that. In the meanwhile, try using this selector as an ignoredNodes value: `"TemplateLiteral > *.expressions"` Not 100% sure it will work, but it should be in the right direction.

**Comment by platinumazure** [comment]
@mercmobily You could use the `indent-legacy` rule if you want indent from ESLint 3.x, while still using latest ESLint. But `indent-legacy` simply doesn't check the indent for a lot of different tokens.

**Comment by mercmobily** [comment]
OK -- will try that. I have to ask though... what are the chances of this getting looked at?
Please realise that it's giving the _wrong_ warnings for template literals; I humbly ask you to consider to turn this into a bug, rather than an enhancement. It makes it painful to use libraries that use template literals as template engines (HyperHTML, Lit-html, etc.)
